### ISSUE TITLE
keepmenu with challenge response?

### ISSUE THREAD

**Issue by n1ete** [firecat53/keepmenu#49]
the title says it all, is there any way to integrate challenge response in keepmenu? or is this already working?
thanks

**Comment by firecat53** [comment]
No, it's not implemented. I don't have a Yubikey or other similar device right now to develop with. I'm also no crypto expert! In addition, one of my goals is to maintain database compatibility between platforms and keepass versions. [This KeepassXC FAQ answer](https://keepassxc.org/docs/#faq-yubikey-incompatible) makes me think that could potentially be an issue. PRs welcome if someone has the knowledge and a good plan for implementation.

**Comment by jrbp** [comment]
I've been looking in to using a yubikey with my database and keepmenu support is the main thing I'd miss (thank you for your work on it!).

Since this issue was opened it seems many keepass compatible apps (Keepass2Android, KeepassDX, KeePassium) have settled on supporting the KeePassXC approach for challenge response. Unlike KeePassXC, KeePass 2 doesn't support YubiKeys directly, but through a plugin, the most popular seeming to be KeeChallenge. KeePassium has [a page with reasons KeeChallenge isn't supported](https://support.keepassium.com/kb/yubikey-keechallenge/).

I'm not confident I'm the right person to do it, but would a PR supporting only the KeePassXC approach be welcome?

**Comment by firecat53** [comment]
PR's always welcome! Thanks for the interest.
